utils/node_local_test.py

```diff
@@ -2385,77 +2420,204 @@ def test_cont_clone(self):
         destroy_container(self.conf, self.pool.id(), container)
 # pylint: enable=too-many-public-methods
 
+class nlt_stdout_wrapper():
+    """Class for capturing stdout from threads"""
+
+    def __init__(self):
+        self._stdout = sys.stdout
+        self._outputs = {}
+        sys.stdout = self
+
+    def write(self, value):
+        """Print to stdout.  If this is the main thread then print it, always save it"""
+
+        thread = threading.current_thread()
+        if not thread.daemon:
+            self._stdout.write(value)
+        thread_id = thread.ident
+        try:
+            self._outputs[thread_id] += value
+        except KeyError:
+            self._outputs[thread_id] = value
```

Comment:
At first this doesn't seem thread safe, but maybe it's okay because we use `thread_id` as the index. But then, I'm not sure if adding new keys to a dictionary is thread safe while another thread is reading/modifying a different key

Reply:
I have to assume it is, although if each thread was using a different key then I don't think it would be.

Reply:
Some quick Googling makes me think this is okay